spec/models/reservation_spec.rb

```diff
@@ -26,4 +26,38 @@
   it 'validates raise error for invalid status' do
     expect { create(:reservation, status: 'invalid') }.to raise_error(ArgumentError, "'invalid' is not a valid status")
   end
+
+  describe 'associations' do
+    it "ensure to don't create a reservation with an unavailable room" do
+      _available_room = create(:available_double_room)
+      unavailable_room = create(:unavailable_double_room)
+      reservation = create(:reservation, room_id: unavailable_room.id)
+
+      expect(reservation.room.status).to eq('available')
+    end
```

Comment:
The description of the test is very hard to understand with double negatives. As a unit test, it is unclear to me what "method" you are testing against?